shell/platform/darwin/ios/framework/Source/FlutterEngine.mm

```diff
@@ -986,6 +986,37 @@ - (void)flutterTextInputView:(FlutterTextInputView*)textInputView
                               arguments:@[ @(client) ]];
 }
 
+- (void)flutterTextInputViewDidResignFirstResponder:(FlutterTextInputView*)textInputView {
+  // Platform view's first responder detection logic
+  //
+  // All text input widgets (e.g. EditableText) are backed by a dummy UITextInput view
+  // in the text input plugin. When this dummy UITextInput view resigns first responder,
+  // check if any platform view becomes first responder. If any platform view becomes
+  // first responder, send a "viewFocused" channel message to inform the framework to un-focus
+  // the previously focused text input.
+  //
+  // Caveat:
+  // 1. This detection logic does not cover the scenario when a platform view becomes
+  // first responder without any flutter text input resigning its first responder status
+  // (e.g. user tapping on platform view first). For now it works fine because there can only be
```

Comment:
Just curious:
Are we expecting an UIKit update that will introduce multiple first responders?

Reply:
I should probably rephrase it clearly - I am not expecting UIKit change. What I meant to say is that right now the text input plugin does not need to keep track of the platform view focus state, but in the future we may need to track it (not because of UIKit change, but for potential other unknown reasons)